src/System.Management.Automation/engine/NativeCommandProcessor.cs

```diff
@@ -403,8 +411,15 @@ internal NativeCommandParameterBinderController NativeParameterBinderController
         /// </summary>
         internal override void Prepare(IDictionary psDefaultParameterValues)
         {
+            if (_isPreparedCalled)
+            {
+                return;
+            }
+
             _isPreparedCalled = true;
 
+            DownStreamNativeCommand?.Prepare(psDefaultParameterValues);
```

Comment:
This can probably be removed. Initially I wasn't lazily retrieving the stdout stream so the downstream native command had to be prepared before the upstream. I believe I've fixed that